### PR TITLE
Add description to VD E2E tests

### PR DESCRIPTION
**Pull request by MARCOSD4**
# Description

This PR aims to add a description to all the tests within the Vulnerability Detector E2E tests, as it is useful to understand what the test is doing and what results are expected.

## Testing performed

One of the tests has been executed, with the following environment https://ci.wazuh.info/job/Wazuh_QA_environment/1029/, to verify that the missing test description appears correctly in the final report:

image

[report5110.zip](https://github.com/wazuh/wazuh-qa/files/14757253/report5110.zip)

### CODE REVIEW

File: tests/end_to_end/test_vulnerability_detector/test_vulnerability_detector.py [hunk 1]
```diff
@@ -556,7 +556,39 @@ def get_results(self):
...
+            - body: the body of the test case, which contains the tasks to be executed
+            - teardown: the teardown within the test cases, if any
+            - setup: test setup results, to check if the hosts are setup correctly
+            - host_manager:
+                type: fixture
+                brief: Get the host manager of the environment            
+            - get_results: fixture to get the results of global class tests
+
+        assertions:
+            - Verify that all the hosts are properly setup.
```
Comment (juliamagan) [inline tests/end_to_end/test_vulnerability_detector/test_vulnerability_detector.py:583]: It also checks whether vulnerabilities appear, disappear, or remain and whether alerts also appear.

File: tests/end_to_end/test_vulnerability_detector/test_vulnerability_detector.py [hunk 2]
```diff
@@ -556,7 +556,39 @@ def get_results(self):
...
+
+        cases: None
```
Comment (juliamagan) [inline tests/end_to_end/test_vulnerability_detector/test_vulnerability_detector.py:585]: The cases are all those in yaml, and appear between `[]` in the report.

**Review by juliamagan — CHANGES_REQUESTED** [review]
GJ, but some changes are required. In addition, remember to add the changelog entry.

File: tests/end_to_end/test_vulnerability_detector/test_vulnerability_detector.py (hunk 1 above)
Comment (MARCOSD4) [reply tests/end_to_end/test_vulnerability_detector/test_vulnerability_detector.py:583]: Fixed in https://github.com/wazuh/wazuh-qa/pull/5151/commits/af7ba540763c3f445479f379b4dbed5d40b4ce23

File: tests/end_to_end/test_vulnerability_detector/test_vulnerability_detector.py (hunk 2 above)
Comment (MARCOSD4) [reply tests/end_to_end/test_vulnerability_detector/test_vulnerability_detector.py]: Fixed in https://github.com/wazuh/wazuh-qa/pull/5151/commits/af7ba540763c3f445479f379b4dbed5d40b4ce23

File: tests/end_to_end/test_vulnerability_detector/test_vulnerability_detector.py [hunk 3]
```diff
@@ -556,7 +556,49 @@ def get_results(self):
 
     @pytest.mark.parametrize('preconditions, body, teardown', complete_list, ids=list_ids)
     def test_vulnerability_detector_scans_cases(self, setup_vulnerability_tests, request, preconditions, body, teardown, setup,
-                                                host_manager, get_results):
+                                                host_manager, get_results):        
+        """
+        description: validates the Vulnerability Detector's ability to detect new vulnerabilities in the environment for each of the defined cases.
```
Comment (Rebits) [inline tests/end_to_end/test_vulnerability_detector/test_vulnerability_detector.py:561]:
```suggestion
        description: Validates the Vulnerability Detector's ability to detect new vulnerabilities in the environment for each of the defined cases.
```

File: tests/end_to_end/test_vulnerability_detector/test_vulnerability_detector.py [hunk 4]
```diff
@@ -556,7 +556,49 @@ def get_results(self):
...
+
+        This is the main test. It is in charge of executing each of the cases defined in test_vulnerability.yaml, 
+        checking if all agents generate the corresponding vulnerabilities and collecting the results.
```
Comment (Rebits) [inline tests/end_to_end/test_vulnerability_detector/test_vulnerability_detector.py:564]:
```suggestion
This test evaluates the effectiveness of the Vulnerability Detector in real-world scenarios, focusing on the installation, removal, or upgrade of various vulnerable and non-vulnerable packages in the environment. It ensures that all agents generate the expected vulnerabilities and associated alerts.
```

File: tests/end_to_end/test_vulnerability_detector/test_vulnerability_detector.py [hunk 5]
```diff
@@ -556,7 +556,49 @@ def get_results(self):
...
+
+        tier: 1
```
Comment (Rebits) [inline tests/end_to_end/test_vulnerability_detector/test_vulnerability_detector.py:566]:
```suggestion
        tier: 0
```

File: tests/end_to_end/test_vulnerability_detector/test_vulnerability_detector.py [hunk 6]
```diff
@@ -556,7 +556,49 @@ def get_results(self):
...
+
+        parameters:
+            - setup_vulnerability_tests:
+                type: fixture
+                brief: Setup the environment to proceed with the testing
```
Comment (Rebits) [inline tests/end_to_end/test_vulnerability_detector/test_vulnerability_detector.py:571]: We should respect this structure for all the parameters

File: tests/end_to_end/test_vulnerability_detector/test_vulnerability_detector.py [hunk 7]
```diff
@@ -556,7 +556,49 @@ def get_results(self):
...
+            - body: the body of the test case, which contains the tasks to be executed
+            - teardown: the teardown within the test cases, if any
+            - setup: test setup results, to check if the hosts are setup correctly
+            - host_manager:
+                type: fixture
+                brief: Get the host manager of the environment            
+            - get_results: fixture to get the results of global class tests
+
+        assertions:
+            - Verify that all the hosts are properly setup.
```
Comment (Rebits) [inline tests/end_to_end/test_vulnerability_detector/test_vulnerability_detector.py:583]: Missing assertions

**Review by Rebits — CHANGES_REQUESTED** [review]
GJ. Minor changes are required

File: tests/end_to_end/test_vulnerability_detector/test_vulnerability_detector.py (hunk 7 above)
Comment (Rebits) [reply tests/end_to_end/test_vulnerability_detector/test_vulnerability_detector.py:583]: Several other assertions are implicitly verified, such as ensuring that remote operations in the environment correctly produce results, such as the installation or removal of packages. However, these assertions will become clearer following the development outlined in https://github.com/wazuh/wazuh-qa/issues/4914. Therefore, I recommend temporarily hiding this suggestion.

File: tests/end_to_end/test_vulnerability_detector/test_vulnerability_detector.py (hunk 3 above)
Comment (MARCOSD4) [reply tests/end_to_end/test_vulnerability_detector/test_vulnerability_detector.py]: Fixed in https://github.com/wazuh/wazuh-qa/pull/5151/commits/b1da35a1fb87428a5059f7574e32fed607aa28a7

File: tests/end_to_end/test_vulnerability_detector/test_vulnerability_detector.py (hunk 4 above)
Comment (MARCOSD4) [reply tests/end_to_end/test_vulnerability_detector/test_vulnerability_detector.py]: Fixed in https://github.com/wazuh/wazuh-qa/pull/5151/commits/b1da35a1fb87428a5059f7574e32fed607aa28a7

File: tests/end_to_end/test_vulnerability_detector/test_vulnerability_detector.py (hunk 5 above)
Comment (MARCOSD4) [reply tests/end_to_end/test_vulnerability_detector/test_vulnerability_detector.py]: Fixed in https://github.com/wazuh/wazuh-qa/pull/5151/commits/b1da35a1fb87428a5059f7574e32fed607aa28a7

File: tests/end_to_end/test_vulnerability_detector/test_vulnerability_detector.py (hunk 6 above)
Comment (MARCOSD4) [reply tests/end_to_end/test_vulnerability_detector/test_vulnerability_detector.py:572]: Fixed in https://github.com/wazuh/wazuh-qa/pull/5151/commits/b1da35a1fb87428a5059f7574e32fed607aa28a7

File: CHANGELOG.md [hunk 8]
```diff
@@ -49,6 +49,7 @@ All notable changes to this project will be documented in this file.
 
 ### Fixed
 
+- Fix Vulnerability Detector E2E tests by adding description to all tests([#5151](https://github.com/wazuh/wazuh-qa/pull/5151)) \- (Tests)
```
Comment (juliamagan) [inline CHANGELOG.md:52]:
```suggestion
- Fix Vulnerability Detector E2E tests by adding description to all tests ([#5151](https://github.com/wazuh/wazuh-qa/pull/5151)) \- (Tests)
```

**Review by juliamagan — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 8 above)
Comment (MARCOSD4) [reply CHANGELOG.md]: Fixed in https://github.com/wazuh/wazuh-qa/pull/5151/commits/eea576b2737cb1afb0a13b8dd09801781e6defaf

**Review by Rebits — APPROVED** [review]
LGTM

**Pull request merged**